components/search_bar/search_bar.jsx

```diff
@@ -79,12 +80,45 @@ export default class SearchBar extends React.PureComponent {
         }
     }
 
+    componentWillReceiveProps(nextProps) {
+        const {searchTerms} = nextProps;
+
+        this.setState({
+            visibleSearchHintOptions: this.determineVisibleSearchHintOptions(searchTerms),
+        });
+    }
+
+    determineVisibleSearchHintOptions = (searchTerms) => {
+        let visibleSearchHintOptions = [];
+
+        if (searchTerms.trim() === '') {
+            visibleSearchHintOptions = searchHintOptions;
+        } else {
+            const pretextArray = searchTerms.split(' ');
+            const pretext = pretextArray[pretextArray.length - 1];
+
+            const shouldShowHintOptions = !searchHintOptions.some(({searchTerm}) => searchTerms.toLowerCase().endsWith(`${searchTerm.toLowerCase()} `));
+
+            if (shouldShowHintOptions) {
+                try {
+                    visibleSearchHintOptions = searchHintOptions.filter((option) => {
+                        return new RegExp(pretext, 'ig').test(option.searchTerm) && option.searchTerm.toLowerCase() !== pretext.toLowerCase();
+                    });
+                } catch (err) {
```

Comment:
is `err` needed? Didn't linter complain?

Reply:
Whoops. Will remove